### DISCUSSION TITLE
Background on Nomad Agent removal?

### DISCUSSION THREAD

**Discussion by wchatx** [PrefectHQ/prefect#3575] [General] [answered]
We're exploring adoption of Prefect but we're using Nomad as our container scheduler. I see there was a Nomad Agent at one point but it was removed in #2492. Hoping to find out a little background on why it was removed and if there's any recommendation for how to handle in our case. We came up with a few different ideas but they're all less ideal than a dedicated agent for Nomad.

Thanks!

**Comment by joshmeek** [comment, marked as answer]
Hi @wchatx the Nomad Agent was originally built as a proof of concept at the time. It quickly fell behind active development because we didn't observe any users actively using it as most opted for one of the other options (Local, Docker, Kubernetes, Fargate). We'd be happy to accept a contribution for a new Nomad Agent as I think I have seen one or two others mention in the public Slack channel that they would use it. The addition of this agent can be as straightforward as making a class that inherits from the base `Agent` class and implements the `deploy_flows` function to create flow runs on Nomad. I don't have much knowledge on Nomad outside of that so I don't have any recommendations or best practices on how to accomplish it.
